### PR TITLE
Fix #1482

### LINKED ISSUE #1482: Donate button does not reappear if login is cancelled
## Issue Overview

This would be rare, but if a user clicks Login, then cancels because they need to create an account, the Donate button remains hidden.

## Steps to Reproduce (for bugs)

1. Create form with Registration + Login enabled.
2. View form while logged out.
3. Click "Already have an account? Login"
4. Notice that Donate button diseappers when Login form appears. This is expected since the form setting requires a registration.
5. Click Cancel button below login fields.
6. The Donate button should reappear but it is still hidden, preventing the form from being submitted even if the user fills out registration fields.

Give version: 1.8
WP version: 4.7.2

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This issue will resolve #1482 

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@DevinWalker Please regenerate  `give.all.min.js` while merge this PR.

File: assets/js/frontend/give-ajax.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ jQuery(document).ready(function ($) {
 	 *
 	 * @description: Process the donation submit
 	 */
-	$('body').on('click touchend', 'form.give-form input[type="submit"].give-submit', function (e) {
+	$('body').on('click touchend', 'form.give-form input[name="give-purchase"].give-submit', function (e) {
```
Comment (ravinderk) [inline assets/js/frontend/give-ajax.js]: @DevinWalker Check this. This event was also firing for login button click, so I updated it to more specific ( for donate button)

**Comment by ravinderk** [comment]
@DevinWalker I think for Donate Now button id attribute should be `give-purchase-button-{form_id}` instead of `give-purchase-button` to prevent future js or CSS conflicts.
Can we review all selector of donation form element?

For ref: https://github.com/WordImpress/Give/blob/master/includes/forms/template.php#L1629

Let me know what do you think?

**Comment by DevinWalker** [comment]
@ravinderk we are going to revise the ID selectors in 1.9, not in a minor release. We can update the submit button ID for this release to `give-donate-button-{form_id}`.

**Pull request closed without merging**